src/components/query-search-list/query-search-list.js

```diff
@@ -35,7 +35,7 @@ const QuerySearchList = ({
 
   useEffect(() => {
     fetch(collection.page);
-  }, []);
+  }, [collection.page, fetch]);
```

Comment:
I think this might be causing duplicate requests

https://user-images.githubusercontent.com/19309423/209317381-7c4a34f0-1d9e-45ec-bddf-33bc020e3744.mp4

Reply:
Yes. Thank you for the observation!
I have reverted those dependencies. Please review